### PR TITLE
add postCommentToArticle functionality and will seperate comment-orie…

### PR DESCRIPTION
**Pull request by AlexMSm**
…ntated functionalities to their own test, model and controller files. With tests for bad inputs.

### CODE REVIEW

File: _models/articles.models.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ const db = require("../db/connection");
 const { selectTopics } = require("./topics.models");
 
 exports.selectArticleById = (articleId) => {
-  if (!(articleId > 0)) {
+  if (!(articleId > 0) || !Number.isInteger(Number(articleId))) {
```
Comment (AlexMSm) [inline _models/articles.models.js:6]: Extra protection against floats

File: _models/comments.models.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+/* formatComment = (username) => {
+  console.log(username);
+  return db
+    .query("SELECT username, name FROM users WHERE username = $1;", [username])
+    .then((response) => {
+      return createRef(response.rows, "username", "name");
+    });
+};
+//const userFormat = await formatComment(comment.username);
+//comment["username"] = userFormat[comment.username]; */
```
Comment (AlexMSm) [inline _models/comments.models.js:40]: Originally thought I needed to use the createRef func to insert the name of the user not the username - keeping the code commented out again incase it comes of use.

File: db/data/test-data/users.js [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
-    username: 'lurker',
-    name: 'do_nothing',
+    username: "lurker",
+    name: "do_nothing",
     avatar_url:
-      'https://www.golenbock.com/wp-content/uploads/2015/01/placeholder-user.png'
-  }
+      "https://www.golenbock.com/wp-content/uploads/2015/01/placeholder-user.png",
+  },
 ];
```
Comment (AlexMSm) [inline db/data/test-data/users.js:24]: This was an accident.

File: _models/articles.models.js (hunk 1 above)
Comment (MrMcSnail) [reply _models/articles.models.js:6]: fancy 🕺

File: _models/comments.models.js (hunk 2 above)
Comment (MrMcSnail) [reply _models/comments.models.js:40]: Pop it in another file that is git ignored. Main branch should be sacrosanct. In continuous development your remote main might literally be to code that gets pushed out to the client's devices

File: db/data/test-data/users.js (hunk 3 above)
Comment (MrMcSnail) [reply db/data/test-data/users.js:24]: Linters eh? 🤷

Comment (MrMcSnail) [reply db/data/test-data/users.js:24]: Make sure you're being selective with your git staging

**Review by MrMcSnail — APPROVED** [review]
Crack on and merge!

**Pull request merged**